app/src/modules/fota/fota.c

```diff
@@ -130,59 +159,97 @@ static void state_initialized_run(void *o)
 		const enum cloud_status status = MSG_TO_CLOUD_STATUS(state_object->msg_buf);
 
 		if (status == CLOUD_CONNECTED_READY_TO_SEND) {
-			STATE_SET(STATE_POLL_AND_PROCESS);
+			STATE_SET(STATE_WAIT_FOR_TRIGGER);
 		}
 	}
 }
 
-static void state_poll_and_process_run(void *o)
+static void state_wait_for_trigger_run(void *o)
 {
 	struct s_object *state_object = o;
 
 	if (&TRIGGER_CHAN == state_object->chan) {
-		int err;
 		const enum trigger_type trigger_type = MSG_TO_TRIGGER_TYPE(state_object->msg_buf);
-		enum fota_status fota_status = FOTA_STATUS_PROCESSING_START;
 
-		if (trigger_type != TRIGGER_FOTA_POLL) {
-			return;
+		if (trigger_type == TRIGGER_FOTA_POLL) {
+			STATE_SET(STATE_POLL_AND_PROCESS);
 		}
+	}
 
-		/* Tell the rest of the app not to disturb */
-		err = zbus_chan_pub(&FOTA_STATUS_CHAN, &fota_status, K_NO_WAIT);
-		if (err) {
-			LOG_ERR("zbus_chan_pub, error: %d", err);
-			SEND_FATAL_ERROR();
+	if (&CLOUD_CHAN == state_object->chan) {
+		const enum cloud_status status = MSG_TO_CLOUD_STATUS(state_object->msg_buf);
+
+		if (status == CLOUD_CONNECTED_PAUSED) {
+			STATE_SET(STATE_INITIALIZED);
 		}
+	}
+}
 
-		LOG_DBG("Polling for FOTA updates");
+static void state_poll_and_process_entry(void *o)
+{
+	struct s_object *state_object = o;
+	/* Notify the rest of the system that FOTA processing is starting */
+	enum fota_status fota_status = FOTA_STATUS_PROCESSING_START;
 
-		err = nrf_cloud_fota_poll_process(&state_object->fota_ctx);
-		if (err == -EBUSY) {
-			/* The reboot handler was called, but the reboot was not performed as it
-			 * be handled in the reboot pending state to let the rest of the system
-			 * know that a reboot is pending for graceful shutdown.
-			 */
-			STATE_SET(STATE_REBOOT_PENDING);
+	int err = zbus_chan_pub(&FOTA_STATUS_CHAN, &fota_status, K_NO_WAIT);
+	if (err) {
+		LOG_ERR("zbus_chan_pub, error: %d", err);
+		SEND_FATAL_ERROR();
+	}
 
-			return;
-		}
+	/* Start the FOTA processing */
+	err = nrf_cloud_fota_poll_process(&state_object->fota_ctx);
+	if (err) {
+		enum priv_fota_evt evt = FOTA_PRIV_PROCESSING_COMPLETED;
 
-		if (err && err != -EAGAIN) {
+		if (err != -EAGAIN) {
 			LOG_ERR("nrf_cloud_fota_poll_process failed: %d", err);
 		}
 
-		/* FOTA processing is completed, notify on the channel */
-		fota_status = FOTA_STATUS_PROCESSING_DONE;
-
-		err = zbus_chan_pub(&FOTA_STATUS_CHAN, &fota_status, K_NO_WAIT);
+		err = zbus_chan_pub(&PRIV_FOTA_CHAN, &evt, K_NO_WAIT);
 		if (err) {
 			LOG_ERR("zbus_chan_pub, error: %d", err);
 			SEND_FATAL_ERROR();
 		}
 	}
 }
 
+static void state_poll_and_process_run(void *o)
+{
+	struct s_object *state_object = o;
+
+	if (&PRIV_FOTA_CHAN == state_object->chan) {
+		const enum priv_fota_evt evt = *(const enum priv_fota_evt *)state_object->msg_buf;
+
+		switch (evt) {
+		case FOTA_PRIV_PROCESSING_COMPLETED: {
+			STATE_SET(STATE_WAIT_FOR_TRIGGER);
```

Comment:
Why do we need to wait for a trigger when the processing is completed?

Reply:
When processing is done (with or without a job vailable from the cloud), there could be a new job queued up, so we go back to wait for a new trigger